client/server/server.go

```diff
@@ -593,7 +593,27 @@ func (s *Server) Down(_ context.Context, _ *proto.DownRequest) (*proto.DownRespo
 	state := internal.CtxGetState(s.rootCtx)
 	state.Set(internal.StatusIdle)
 
-	return &proto.DownResponse{}, nil
+	maxWaitTime := 5 * time.Second
+	startTime := time.Now()
+
+	engine := s.connectClient.Engine()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return &proto.DownResponse{}, nil
+		default:
+			if !engine.IsWGIfaceUp() {
+				return &proto.DownResponse{}, nil
+			}
+
+			if time.Since(startTime) > maxWaitTime {
```

Comment:
I think it's more idomatic to use `time.After` here and include the returned channel in the `select`, unless you still need to check `if !engine.IsWGIfaceUp()` in any case?